closed/src/java.base/share/classes/openj9/internal/security/RestrictedSecurityProperties.java

```diff
@@ -0,0 +1,697 @@
+        }
+
+        if (securityProps == null) {
+            throw new RuntimeException(
+                    "Restricted security mode initialization error, call getInstance with variables first.");
+        }
+
+        try {
+
+            // Print out the Help and Audit info.
+            if (userSecurityHelp) {
+                printHelp();
+                if (userSecurityNum == 0) {
+                    if (debug != null) {
+                        debug.println("Print out the help info and exit.");
+                    }
+                    System.exit(0);
+                }
+            }
+            if (userSecurityAudit) {
+                listAudit();
+                if (userSecurityNum == 0) {
+                    if (debug != null) {
+                        debug.println("Print out the audit info and exit.");
+                    }
+                    System.exit(0);
+                }
+            }
+
+            // Load the restricted security providers from java.security properties.
+            initProviders();
+            // Load the restricted security properties from java.security properties.
+            initProperties();
+            // Load the restricted security provider constraints from java.security properties.
+            initConstraints();
+
+            // Print out the Trace info.
+            if (userSecurityTrace) {
+                listTrace();
+            }
+
+            if (debug != null) {
+                debug.println("Initialized restricted security mode.");
+            }
+        } catch (Exception e) {
+            if (debug != null) {
+                debug.println("Unable to initialize restricted security mode.");
+            }
+            e.printStackTrace();
+        }
+    }
+
+    /**
+     * Load restricted security provider.
+     */
+    private void initProviders() {
+
+        if (debug != null) {
+            debug.println("Loading restricted security providers.");
+        }
+
+        for (int pNum = 1;; ++pNum) {
+            String providerInfo = securityProps
+                    .getProperty(propsPrefix + ".jce.provider." + pNum);
+
+            if (providerInfo == null || providerInfo.trim().isEmpty()) {
+                break;
+            }
+
+            if (!areBracketsBalanced(providerInfo)) {
+                new RuntimeException("Restricted security provider format is inccorect: " + providerInfo)
+                        .printStackTrace();
+                System.exit(1);
+            }
+
+            int pos = providerInfo.indexOf('[');
+            String providerName = (pos < 0) ? providerInfo.trim() : providerInfo.substring(0, pos).trim();
+            // Provider with argument (provider name + optional argument).
+            providers.add(pNum - 1, providerName);
+
+            pos = providerName.indexOf(' ');
+            providerName = (pos < 0) ? providerName.trim() : providerName.substring(0, pos).trim();
+            // Provider without argument.
```

Comment:
These lines need to be added here as well: the package should not be part of the provider’s “short name”, otherwise the comparison on line 433 fails.
```
// Remove the provider's class package name.
pos = providerName.lastIndexOf('.');
providerName = (pos > 0) ? providerName.substring(pos + 1, providerName.length()) : providerName;
```